src/niftyone/figures/generator.py

```diff
@@ -0,0 +1,91 @@
+"""Generator classes for different views."""
+
+import logging
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Callable, Generic, TypeVar
+
+import nibabel as nib
+import pandas as pd
+from bids2table import BIDSEntities, BIDSTable
+from matplotlib.figure import Figure
+from PIL.Image import Image
+
+import niclips.image as noimg
+
+generator_registry: dict[str, type["ViewGenerator"]] = {}
+
+T = TypeVar("T", bound="ViewGenerator")
+
+
+def register(name: str) -> Callable[[type[T]], type[T]]:
+    """Function to register generator to registry."""
+
+    def decorator(cls: type[T]) -> type[T]:
+        generator_registry[name] = cls
+        return cls
+
+    return decorator
+
+
+def create_generators(
+    config: dict[str, Any],
+) -> list["ViewGenerator"]:
+    """Function to create generators dynamically from config."""
+    generators = []
+
+    for settings in config.values():
```

Comment:
Here the keys of `config` are discarded. Should we keep them and return a `dict[str, "ViewGenerator"]`?

Reply:
Hmm, I wasn't sure what to use the keys here for since they seem arbitrary and are really only used to group the query / views. I do like the idea of keeping them around more I think about it.

Reply:
I think I am inclined to leave this as is here unless we have a use case for the keys.